### PR TITLE
Add GOVERNANCE.md document

### PR DESCRIPTION
**Pull request by Daeraxa**
The idea of this document is to somewhat clarify the organizational structure behind the org and have some place to reference in case it is needed or asked. For such a new project we have a lot of active contributors and by virtue of the project itself we inherited a bunch of stuff to deal with right off the bat rather than being able to grow it organically which does make keeping on top of things a little more difficult.

My main goal with this is not to add anything we aren't already doing (with one exception) nor is it to try to take the fun out of the project with strict rules and bureaucracy. It is designed so that we can have somewhere to actually come up with plans for when we have to deal with difficult situations that nobody otherwise wants to bring up.

For example the one section I added which we don't have precedence for is "removal of roles/members" which is something that has come up in discussion before for those who are still part of the GitHub org but aren't currently participating in the project.
I think this section needs more fleshing out with something we would have to decide on - for example what duration of inactivity or low contribution before we consider somebody "inactive"?
The idea here isn't to punish, it is to make administration easier and grant privileges and access only as required to prevent any damage (malicious or unintentional) to the org and team.

Much of this was based on the questions and guidance in https://opensource.com/article/20/5/open-source-governance. 

Very happy to take comments on this as I understand it could well be controversial and may need rewording in more than a few places or we may decide to forgo it entirely and place an abridged version in POLICY.md (or nothing at all).

### CODE REVIEW

**Comment by confused-Techie** [comment]
I appreciate you getting to this one! Overall, this is a super well written document, and very clearly defines what little governance policies we have and need.

I love the care taken on writing the new section of policies, but totally agree that we need to put a numeric value there to define inactivity. And whatever timeframe chosen I feel like has to be measured in months. Such as maybe 6 months? Or 3?

It is really hard to say, but I'd be leaning towards the three personally. But curious to hear if you have any thoughts?

**Comment by Daeraxa** [comment]
Thanks for the kind words, had a moment of rare inspiration in the early hours of the morning.
Yeah I think 3 months is acceptable, of course we can still use our discretion and we wouldn't be beholden to follow it to the letter - for example making a random comment every few weeks wouldn't be 'inactive' by the strictest definition but is clearly not being an active part of the project.

**Review by Spiker985 — APPROVED** [review]

**Review by confused-Techie — APPROVED** [review]
I'll go ahead and give this one an approval assuming we add 3 months as the inactivity period somewhere. Otherwise I'd be interested to hear any other thoughts from the rest of the team

**Comment by Daeraxa** [comment]
I can add the 3 months onto this branch if that suits? e.g. 
```diff
- If a member of a team or role becomes inactive for an extended period of time without a
+ If a member of a team or role becomes inactive for over three months without a
```

**Comment by confused-Techie** [comment]
> I can add the 3 months onto this branch if that suits? e.g.
> 
> ```diff
> - If a member of a team or role becomes inactive for an extended period of time without a
> + If a member of a team or role becomes inactive for over three months without a
> ```

Yeah that could be could, maybe just add an `about three...` for some leniency

**Comment by Daeraxa** [comment]
> Yeah that could be could, maybe just add an `about three...` for some leniency

My only issue with `around` is that it could count as the reverse, say 2 and a half months.
Either way it is still up to our discretion as the next steps at those three months are to attempt to contact followed by a vote phase so it will always be longer than that anyway.

**Comment by confused-Techie** [comment]
> > Yeah that could be could, maybe just add an `about three...` for some leniency
> 
> My only issue with `around` is that it could count as the reverse, say 2 and a half months. Either way it is still up to our discretion as the next steps at those three months are to attempt to contact followed by a vote phase so it will always be longer than that anyway.

Ahh I see, this looks fine then to me

File: GOVERNANCE.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+- Packages - Works on our core packages
+- Developer - Just fills in anywhere that they want
+
+##### Operating Systems
+- Linux
+- MacOS
+- Windows
+
+##### Other roles
+- Community Packages - join if you're into making packages for us
```
Comment (mauricioszabo) [inline GOVERNANCE.md:80]:
```suggestion
- Community Packages - join if you're into making packages for the editor
```

Comment (Daeraxa) [reply GOVERNANCE.md:80]: That bit was taken verbatim from discord. I mean, I agree with the change but I think it should be changed there too.

Comment (confused-Techie) [reply GOVERNANCE.md]: Great catch there, this does sound better

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+### Removing members/roles
+
+This is not an action that the organization wants to perform in an ideal circumstances but this is something we do require some reference for so that people do not feel unfairly treated.
+There are a number of circumstances where this might need to be performed and this section outlines (non-exhaustively) those reasons and justifications.
+
+- **Removal requests**: If a member simply asks to be removed from a team or role for any reason then this should be done as they wish with no votes required.
+
+- **Inactivity**: If a member of a team or role becomes inactive for an extended period of time (3+ months) without any justification or conversation then removal may be required. The member should be first contacted by a another team member and, depending on the outcome, it should then be brought up and voted on as with any other decision. If the vote passes then the member will be removed from these various roles and teams. This vote may be to remove a person entirely or it may be to move to a role with fewer privileges but remain within the organization.
```
Comment (DeeDeeG) [inline GOVERNANCE.md:100]: IMO the criterion isn't "inactive ... _**without ... justification or conversation**_", it's just inactivity. We are not removing access to make a judgment in the person, we are doing it as a routine security measure, and I epxect our posture toward re-adding people to be pretty open-minded and easy-going, though we may require activity again just like a new member? (Not to judge but to be cautious and disincentivize account takeovers by any malicious actors out there, and establish the person controlling the account is the same person.)

Likewise, the above where "This is not an action that the organization wants to perform in an ideal circumstances", (I see how that sentiment applies for conduct removals sure, but...) with my security hat on, rearding the timeout removals yes, we do want to get over the intuition hump of this being a thing with heightened emotions involved, and be doing it routinely and without partiality and yet also without any sort of judgment or social friction, I think for safety we need to be technocratic in just this one area and "just do it".

(Not planning this, posing this as a hypothetical, mind:) If I go on vacation for like several months, I hope there will be someone willing to review my application to come back on with perms, and most of what needs doing around can be done (PRs, issues, etc.) without _any_ internal permissions. But yeah, I want the heat off my account as a target for breaches if I'm genuinely away for an extended period of time. IMO that should be the framing.

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+Sometimes it may be necessary to modify the organizational structure as defined in this document.
+This would be done in the same way as any other change as per our [Policies](https://github.com/pulsar-edit/.github/blob/main/POLICY.md) with proposals and a vote.
+If the vote passes then any new team or role additions can be performed in the same way as this document describes.
+
+## Other roles and responsibilities
+
+Not every single role or responsibility has been detailed in this document. Some responsibilities are handled as and when required and to the person or people most suited for it.
+Such areas include access details and administration of various tools and services used by the organization, social media account access, fund/donation access and expenses approval and organization secrets access.
+The above policies regarding adding or removing people to these specific privileges still apply.
```
Comment (DeeDeeG) [inline GOVERNANCE.md:114]: In all honesty, I think we don't in practice _have_ policies we go by on this stuff. We have not been voting, for instance, on who has access to the non-GitHub services we use. Cirrus is implicit via GitHub perms. etc.

This section feels "docs before the policy", "cart before horse", and I don't like being critical, but it doesn't sit right telling people one thing when I full well expect us to do another.

File: GOVERNANCE.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+## Team member & role additions/removals
+
+Team/role administration and membership decision making is mostly covered by our [Policies](https://github.com/pulsar-edit/.github/blob/main/POLICY.md) but this section details the criteria and justifications for adding or removing team members and/or roles.
+
+### Adding members/roles
+
+Adding team members is performed in much the same way as any other decision (see [Policies](https://github.com/pulsar-edit/.github/blob/main/POLICY.md)) by proposals being made to add somebody to a team or role and a vote then taking place on that proposal.
+If the vote passes without any significant disagreements then the member can be added to the relevant team or role.
+There is no strict set of criteria for addition but generally people who are active in the community and project for an extended period of time, have contributed and added value, and share in the organization's goals & vision can be considered.
```
Comment (DeeDeeG) [inline GOVERNANCE.md:91]: I'm not sure we had formalized that adding members goes to a vote, though now that I think about it, it has been done _most_ of the times.

Exceptions were the somewhat haphazard early days when it wasn't clear when/if we were going to branch off from atom-community, and we needed some people with permissions just to have people with permissions and to get the ball rolling on Pulsar (not just PRs and discussions at atom-community org).

But this does feel like putting it in writing, which it is. Just noting it. Mixed feelings, since feels _mildly_ "documentation before policy" "cart before horse", but I'm leaning toward this _does_ feel like it is de-facto policy and I seem to recall informal agreement on this one. Just this makes it somewhat formal.

File: GOVERNANCE.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
+# Governance
+
+This document is designed to document and clarify the governance of the Pulsar (Pulsar-Edit) organization and teams in an open and transparent way.
+The intent is not for this to be a rigid set of rules but this does allow us to have some written reference as to how things are handled when it comes to the team organization and roles.
+
+Members may belong to more than one team or role and some come hand-in-hand with each other.
+
+We do not work on a strict set of responsibilities with a strict governance model, we mostly resemble the [Do-ocracy](https://communitywiki.org/wiki/DoOcracy) approach to governance without too much in the way of formal or elaborate governance and policies but we do of course have to have some guidelines and agreed conventions to have a working organization.
```
Comment (DeeDeeG) [inline GOVERNANCE.md:8]: 👍 Do-ocracy sounds about right, that is indeed how we've been doing it at least.

**Review by DeeDeeG — COMMENTED** [review]
I'm not sure how much of this rises to being actionable, it may end up just being commentary or food for thought.

But these are my thoughts upon reading.

File: GOVERNANCE.md (hunk 2 above)
Comment (DeeDeeG) [reply GOVERNANCE.md:100]: So, I was thinking about this in the abstract before, but when it comes down to actually remove someone (remove someone's permissions, but it does feel personal), in practice I think it's going to be difficult to do that. We all know each-other pretty well at this stage, where we've vetted one-another. So of course it's going to feel personal to remove permissions.

I guess we do in that hypothetical scenario try and talk it out, and if we _can't reach_ the person we need to defer to "just doing it", even if it's actually a little uncomfortable.

I try to shield myself from that kind of sentiment, knowing it's for security, but we're all human, and just trying to technocrat things away doesn't really work a lot of the time nor does it even really dismiss the feelings, in all honestly. It just attempts to puts them out of sight out of mind.

**In summary**: yeah, I guess it's not something we can realistically do without some feelings about it. So the document has kind of hit the mark as-written.

File: GOVERNANCE.md (hunk 4 above)
Comment (Daeraxa) [reply GOVERNANCE.md:91]: Sorry it took so long to come back to this, not sure what happened.
So yeah, this one I think has indeed been *de-facto* policy, the intial grouping was one thing but this has been the process since I joined (02/08) at the very least, including raising access levels.

File: GOVERNANCE.md (hunk 3 above)
Comment (Daeraxa) [reply GOVERNANCE.md:114]: > We have not been voting, for instance, on who has access to the non-GitHub services we use.

No but I think in my opinion it *should* be done - at the very least these should be (where possible) via the admin email account. For those that are scoped around "teams" type accounts then we should probably start to be more granular with those permissions.

It might seem overkill to vote on all of them in which case I'm all ears for alternative wording here.

**Comment by DeeDeeG** [comment]
I finally put together a proper marked up draft 2 of this for discussion purposes, after all this time, sorry for the delay.

My draft/marked up version of the document here, with notes inline:
- Document view: https://github.com/DeeDeeG/pulsar-edit.github/blob/deedeeg-add-governance-draft-2/GOVERNANCE.md
- Diff view: https://github.com/DeeDeeG/pulsar-edit.github/commit/dde5e070193880aa53ee4b834e199716b744e0b1

Hope this can help move things forward, and these are just suggestions as I was reading over it and marking it up with changes I thought to make.

Best,
\- DeeDeeG
